pegasus/router.rb

```diff
@@ -410,7 +414,7 @@ def all_documents
     end
 
     def resolve_document(uri)
-      extnames = settings.non_static_extnames
+      extnames = settings.non_static_extnames + [".#{request.locale}.md"]
```

Comment:
The append here creates a new array which is stored in `extnames`, which is important because we don't want to modify `settings.non_static_extnames` in-place upon every resolve.